platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_pensando_util.py

```diff
@@ -0,0 +1,311 @@
+        if os.path.exists(component_file):
+            data = json.load(open(component_file))
+            component = data["chassis"][list(data["chassis"].keys())[0]]
+        data_dict = {}
+        data_dict["chassis"] = {}
+        data_dict["chassis"][chassis_name] = component
+        with open(component_file, "w") as json_file:
+            json.dump(data_dict, json_file, indent=4)
+    except:
+        pass
+
+def read_qsfp_eeprom_helper(i2c_num,reg):
+    eeprom_byte_array = APIHelper().read_eeprom_qsfp(i2c_num,reg)
+    api_helper_platform = apiHelper.get_platform()
+    platform = api_helper_platform if api_helper_platform != None else "arm64-elba-asic-r0"
+    eeprom_file = f"/usr/share/sonic/device/{platform}/qsfp-eeprom-{i2c_num}.bin"
+    with open(eeprom_file, "wb") as file:
+        file.write(eeprom_byte_array)
+
+def read_qsfp_eeprom():
+    cmd = "cpldapp -r {}".format(QSFP_STAT_CTRL_CPLD_ADDR)
+    try:
+        output = APIHelper().runCMD(cmd)
+        reg_val = int(output, 16)
+        if ((reg_val >> 4) & 1) > 0:
+            read_qsfp_eeprom_helper(1, "0x50")
+        if ((reg_val >> 4) & 2) > 0:
+            read_qsfp_eeprom_helper(2, "0x50")
+    except:
+        pass
+
+def stop_disable_telemetry_service():
+    cmd = "systemctl stop telemetry.service"
+    run_cmd(cmd)
+
+    cmd = "systemctl disable telemetry.service"
+    run_cmd(cmd)
+
+def stop_disable_mgmt_framework_service():
+    cmd = "systemctl stop mgmt-framework.service"
+    run_cmd(cmd)
+
+    cmd = "systemctl disable mgmt-framework.service"
+    run_cmd(cmd)
+
+def stop_disable_featured_service():
+    cmd = "systemctl stop featured.service"
+    run_cmd(cmd)
+
+    cmd = "systemctl disable featured.service"
+    run_cmd(cmd)
+
+def rename_oob_mnic0_to_eth0():
+    cmd = "dhclient oob_mnic0"
+    run_cmd(cmd)
+
+    cmd = "ip link property add dev oob_mnic0 altname eth0"
+    run_cmd(cmd)
+
+    cmd = "dhclient -r oob_mnic0"
+    run_cmd(cmd)
+
+    cmd = "dhclient eth0 &"
+    run_cmd(cmd)
+
+def add_dpu_to_package_json():
+    api_helper_platform = apiHelper.get_platform()
+    platform = api_helper_platform if api_helper_platform != None else "arm64-elba-asic-r0"
+    version_path = "/usr/share/sonic/device/{}/VERSION.json".format(platform)
+    package_path = "/var/lib/sonic-package-manager/packages.json"
+    try:
+        version_data = json.load(open(version_path))
+        package_data = json.load(open(package_path))
+        dpu_version = version_data["sw"]["version"]
+        dpu_sha = version_data["sw"]["sha"]
+        dpu_package_data = {}
+        dpu_package_data["repository"] = "docker-dpu-pensando"
+        dpu_package_data["description"] = "Pensando dpu container"
+        dpu_package_data["default-reference"] = dpu_version
+        dpu_package_data["installed-version"] = dpu_version
+        dpu_package_data["installed"] = True
+        dpu_package_data["built-in"] = True
+        dpu_package_data["image-id"] = f"sha:{dpu_sha}"
```

Comment:
All of this should be populated from `files/build_templates/packages.json.j2`, when `docker-dpu-pensando` is built. Why is this needed?